src/java/voldemort/server/VoldemortConfig.java

```diff
@@ -148,6 +148,7 @@
     public static final String PUSH_HA_LOCK_PATH = "push.ha.lock.path";
     public static final String PUSH_HA_LOCK_IMPLEMENTATION = "push.ha.lock.implementation";
     public static final String PUSH_HA_MAX_NODE_FAILURES = "push.ha.max.node.failure";
+    public static final String PUSH_HA_STATE_AUTO_CLEANUP = "push.ha.state.auto.cleanup";
```

Comment:
Well... do you have any suggestions for how we can prove that it is the right thing to do besides actually trying it? Once battle-tested and proven, I'm totally fine with changing this setting's default to true.